### DISCUSSION TITLE
Example for highlighting columns based on row max or min

### DISCUSSION THREAD

**Discussion by sumanth-tccc** [Avaiga/taipy#1790] [Q&A]
How to highlight  cell of a column  in taipy tables for each row max  or min values?

**Comment by sumanth-tccc** [comment]
for the example for the food db provided 

def table_style(state, index, row):
    return "blue-cell" if row.Category == "Dessert" else ""

def highlight_column_cells(state, value, index, row, column_name):
    if column_name == 'Calories' and row.Category == 'Dessert':
        return "blue-cell"
    return ""

def simple_highlight(state, value, index, row, column_name):
    return "blue-cell"  # Apply to all cells as a test


table_properties = {
    "class_name": "rows-bordered",
    "filter": True,
    "on_edit": food_df_on_edit,
    "on_delete": food_df_on_delete,
    "on_add": food_df_on_add,
    "group_by[Category]": True,
    "apply[Calories]": "sum",
    "style" : simple_highlight
}


none of the function worked for columns color  assignment

**Reply by FlorianJacta** [reply]
You have access to the *state* in your callback. This means you can get *state.food_df* in it. With this variable, you can get the min/max.

Something along those lines:

```python
def highlight_column_cells(state, value, index, row, column_name):
    if column_name == 'Calories':
        if value == float(state.food_df["Calories"].min()):
            return "blue-cell"
    return ""
```

**Reply by sumanth-tccc** [reply]
The markdown syntax
 <|{food_df}|table|style[Calories] =highlight_column_cells|>

 I have tried the above code with [Calories] and without for the above code. This still did not render a highlight for the value. 

There is no issue for whole row highlighting. 

![image](https://github.com/user-attachments/assets/8128af56-cdd1-419c-91e2-d09b1c93e8b3)
